business_rules/operators.py

```diff
@@ -938,6 +939,68 @@ def is_ordered_by(self, other_value: dict) -> pd.Series:
     def is_not_ordered_by(self, other_value: dict) -> pd.Series:
         return ~self.is_ordered_by(other_value)
 
+    @type_operator(FIELD_DATAFRAME)
+    def target_is_sorted_by(self, other_value: dict) -> pd.Series:
+        """
+         Checking if  SESEQ is  sorted based on SESTDTC within USUBJID
+        """
+
+        target: str = self.replace_prefix(other_value.get("target"))
+        columns = list(other_value["user_defined_columns"].keys())
```

Comment:
I think we should expect "within" to be defined separately from the user defined columns. Also I think the user defined columns can be apart of the comparator so it looks something like this:

```
{"target":"SESEQ",  "within": "USUBJID", "comparator": [{"SESTDTC": "DESC"}, {"SMSTDTC": "DESC"]}
```

This way we can automatically group by the within and then apply the sorts